### PR TITLE
Task#192061: Home page

### PR DESCRIPTION
**Pull request by HaiHaChan**

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -3,82 +3,45 @@
...
+        $topViewed = $this->bookRepository->getTopViewedBook($with);
+        $topReview = $this->bookRepository->getTopReviewBook($with);
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view('index', compact('topInteresting', 'topReview', 'topViewed', 'latestBook', 'bestSharing', 'hotUser', 'officeBooks'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/HomeController.php</br>Line exceeds 120 characters; contains 137 characters

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -3,82 +3,45 @@
...
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Whitespace found at end of line

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -3,82 +3,47 @@
...
+        $topReview = $this->bookRepository->getTopReviewBook($with);
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view(
+            'index',
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Whitespace found at end of line

File: app/Http/Controllers/HomeController.php [hunk 4]
```diff
@@ -3,82 +3,47 @@
...
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view(
+            'index', 
+            compact('topInteresting', 'topReview', 'topViewed', 'latestBook', 'bestSharing', 'hotUser', 'officeBooks'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Http/Controllers/HomeController.php [hunk 5]
```diff
@@ -3,82 +3,47 @@
...
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view(
+            'index',
+            compact('topInteresting', 'topReview', 'topViewed', 'latestBook', 'bestSharing', 'hotUser', 'officeBooks'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Http/Controllers/HomeController.php [hunk 6]
```diff
@@ -3,82 +3,46 @@
...
+        $topViewed = $this->bookRepository->getTopViewedBook($with);
+        $topReview = $this->bookRepository->getTopReviewBook($with);
+        $topInteresting = $this->bookRepository->getTopInterestingBook($with);
+        $latestBook = $this->bookRepository->getLatestBook($with);
+        $bestSharing = $this->bookRepository->getBestSharing($with);
+        $hotUser = $bestSharing['user'];
+        $bestSharing = $bestSharing['books'];
+        $officeBooks = $this->bookRepository->getOfficeBooks($offices, $with);
+        
+        return view('index',
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/HomeController.php [hunk 7]
```diff
@@ -3,82 +3,46 @@
...
+            compact('topInteresting', 'topReview', 'topViewed', 'latestBook', 'bestSharing', 'hotUser', 'officeBooks'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Http/Controllers/HomeController.php (hunk 6 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/HomeController.php (hunk 7 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 8]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            ->orderBy('value', 'desc')
+            ->pluck('book_id')
+            ->take(6);
+        if(!empty($ids)) {
+            $books = $this->model()
+            ->select($dataSelect)
+            ->with($with)
+            ->whereIn('id', $ids)
+            ->get();
+
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 9]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            ->groupBy('user_id')
+            ->orderBy('total', 'desc')
+            ->first();
+        if(!empty($user))
+        {
+            $user = app(User::class)->findOrFail($user->user_id);
+            $ids = app(Owner::class)
+                ->where('user_id', $user->id)
+                ->pluck('book_id')
+                ->take(5);
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 10]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            if(!empty($ids)) {
+                $books = $this->model()
+                    ->with($with)
+                    ->whereIn('id', $ids)
+                    ->get();
+
+                return compact('books', 'user');
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 11]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+        return null;
+    }
+
+   public function getOfficeBooks($offices, $with = [])
+    {
+        $data = collect();
+        foreach ($offices as $key => $value) {
+            $ids = app(Bookmeta::class)->where(['key' => 'in-' . str_slug($value), 'value' => '1'])
+                ->get()
+                ->random(3);
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.IncorrectExact</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Line indented incorrectly; expected 4 spaces, found 3

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 12]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            $data->push([
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceIndent</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Opening brace indented incorrectly; expected 3 spaces, found 4

File: app/Http/Controllers/HomeController.php (hunk 6 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/HomeController.php (hunk 7 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 8 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 9 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 10 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 11 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.IncorrectExact</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Line indented incorrectly; expected 4 spaces, found 3

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 12 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceIndent</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Opening brace indented incorrectly; expected 3 spaces, found 4

File: app/Http/Controllers/HomeController.php (hunk 6 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/HomeController.php (hunk 7 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 8 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 9 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 10 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 11 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.IncorrectExact</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Line indented incorrectly; expected 4 spaces, found 3

File: app/Repositories/Eloquents/BookEloquentRepository.php (hunk 12 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceIndent</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Opening brace indented incorrectly; expected 3 spaces, found 4

File: app/Http/Controllers/HomeController.php (hunk 6 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/HomeController.php (hunk 7 above)
Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/HomeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/HomeController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 13]
```diff
@@ -28,8 +34,8 @@ public function find($id, $with = ['medias','categories'], $dataSelect = ['*'])
         $book = $this->model()->findOrFail($id);
 
         return $this->model()
-        ->with($with)
-        ->findOrFail($id);
+            ->with($with)
+            ->findOrFail($id);
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space between comma and "'categories'"; 0 found

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 14]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            ->groupBy('user_id')
+            ->orderBy('total', 'desc')
+            ->first();
+        if (!empty($user))
+        {
+            $user = app(User::class)->findOrFail($user->user_id);
+            $ids = app(Owner::class)
+                ->where('user_id', $user->id)
+                ->pluck('book_id')
+                ->take(config('views.taking_numb.book_best_sharing'));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 15]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+        return null;
+    }
+
+   public function getOfficeBooks($offices, $with = [])
+    {
+        $data = collect();
+        foreach ($offices as $key => $value) {
+            $ids = app(Bookmeta::class)->where(['key' => 'in-' . str_slug($value), 'value' => '1'])
+                ->get()
+                ->random(config('views.random_numb.book'));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.IncorrectExact</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Line indented incorrectly; expected 4 spaces, found 3

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 16]
```diff
@@ -56,4 +62,111 @@ public function getJson($with = [], $data = [], $dataSelect = ['*'], $attribute
...
+            $data->push([
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceIndent</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Opening brace indented incorrectly; expected 3 spaces, found 4

File: app/Http/Controllers/HomeController.php (hunk 7 above)
Comment (BunnyPi04) [reply app/Http/Controllers/HomeController.php]: cái này đóng ngoặc xuống dòng mới đóng nhé

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 17]
```diff
@@ -41,23 +47,110 @@ public function destroy($id)
...
+            ->groupBy('user_id')
+            ->orderBy('total', 'desc')
+            ->first();
+        if (!empty($user))
+        {
+            $user = app(User::class)->findOrFail($user->user_id);
+            $ids = app(Owner::class)
+                ->where('user_id', $user->id)
+                ->pluck('book_id')
+                ->take(config('views.taking_numb.book_best_sharing'));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 18]
```diff
@@ -41,23 +47,110 @@ public function destroy($id)
...
+        return null;
+    }
+
+   public function getOfficeBooks($offices, $with = [])
+    {
+        $data = collect();
+        foreach ($offices as $key => $value) {
+            $ids = app(Bookmeta::class)->where(['key' => 'in-' . str_slug($value), 'value' => '1'])
+                ->get()
+                ->random(config('views.random_numb.book'));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.IncorrectExact</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Line indented incorrectly; expected 4 spaces, found 3

File: app/Repositories/Eloquents/BookEloquentRepository.php [hunk 19]
```diff
@@ -41,23 +47,110 @@ public function destroy($id)
...
+            $data->push([
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceIndent</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookEloquentRepository.php</br>Opening brace indented incorrectly; expected 3 spaces, found 4

**Pull request merged**